go/inst/instance_dao.go

```diff
@@ -2163,6 +2163,22 @@ func ReadHistoryClusterInstances(clusterName string, historyTimestampPattern str
 
 // RegisterCandidateInstance markes a given instance as suggested for successoring a master in the event of failover.
 func RegisterCandidateInstance(instanceKey *InstanceKey, promotionRule CandidatePromotionRule) error {
+	// Work around for https://github.com/github/orchestrator/issues/39
+	// - set IsCandidate based on promotionRule
+	//
+	// read instance information from the orchestrator backend
+	if instance, found, err := ReadInstance(instanceKey); err == nil && found {
+		// determine what the modifed rule should be
+		newIsCandidate := (promotionRule == MustPromoteRule) || (promotionRule == PreferPromoteRule)
+
+		if instance.IsCandidate != newIsCandidate {
+			log.Debugf("RegisterCandidateInstance: %+v: Changing IsCandidate from %v to %v", instance.Key, instance.IsCandidate, newIsCandidate)
+			instance.IsCandidate = newIsCandidate
+			writeInstance(instance, false, nil)
```

Comment:
The `IsCandidate` field does not get written to the `database_instance` table. The introduced `writeInstance()` call doesn't do anything useful.

IsCandidate is inferred on https://github.com/github/orchestrator/blob/8811c420db892c49b1d0ad34d0cc443344861194/go/inst/instance_dao.go#L810-L811

I think this PR is unfortunately wrong in what it attempts to do.